examples/reader.rs

```diff
@@ -0,0 +1,95 @@
+#[macro_use]
+extern crate serde;
+use futures::TryStreamExt;
+use pulsar::{
+    consumer::{ConsumerOptions, InitialPosition},
+    proto::{MessageIdData, Schema},
+    reader::Reader,
+    Authentication, DeserializeMessage, Payload, Pulsar, TokioExecutor,
+};
+use std::env;
+use tokio::time::{sleep, Duration};
+
+#[derive(Serialize, Deserialize)]
+struct TestData {
+    data: String,
+}
+
+impl DeserializeMessage for TestData {
+    type Output = Result<TestData, serde_json::Error>;
+
+    fn deserialize_message(payload: &Payload) -> Self::Output {
+        serde_json::from_slice(&payload.data)
+    }
+}
+#[tokio::main]
+async fn main() -> Result<(), pulsar::Error> {
+    env_logger::init();
+
+    let addr = env::var("PULSAR_ADDRESS")
+        .ok()
+        .unwrap_or_else(|| "pulsar://127.0.0.1:6650".to_string());
+    let topic = env::var("PULSAR_TOPIC")
+        .ok()
+        .unwrap_or_else(|| "non-persistent://public/default/test".to_string());
+
+    let mut builder = Pulsar::builder(addr, TokioExecutor);
+
+    if let Ok(token) = env::var("PULSAR_TOKEN") {
+        let authentication = Authentication {
+            name: "token".to_string(),
+            data: token.into_bytes(),
+        };
+
+        builder = builder.with_auth(authentication);
+    }
+
+    let pulsar: Pulsar<_> = builder.build().await?;
+
+    let mut reader: Reader<TestData, _> = pulsar
+        .reader()
+        .with_topic(topic)
+        .with_consumer_name("test_reader")
+        .with_options(
```

Comment:
why is the reader example setting the durable option and the initial position? Those should be set already when creating the reader(durable to false, initial position to latest)

Reply:
sure, you're right. it's done